src/newgrf.cpp

```diff
@@ -4853,21 +4853,48 @@ static void NewSpriteGroup(ByteReader *buf)
 					act_group = group;
 					group->version = type;
 					if (type == 0) {
+						group->num_input = 3;
 						for (uint i = 0; i < 3; i++) {
 							group->subtract_input[i] = (int16)buf->ReadWord(); // signed
 						}
+						group->num_output = 2;
 						for (uint i = 0; i < 2; i++) {
 							group->add_output[i] = buf->ReadWord(); // unsigned
 						}
 						group->again = buf->ReadByte();
-					} else {
+					} else if (type == 1) {
+						group->num_input = 3;
 						for (uint i = 0; i < 3; i++) {
 							group->subtract_input[i] = buf->ReadByte();
 						}
+						group->num_output = 2;
 						for (uint i = 0; i < 2; i++) {
 							group->add_output[i] = buf->ReadByte();
 						}
 						group->again = buf->ReadByte();
+					} else if (type == 2) {
+						group->num_input = buf->ReadByte();
+						if (group->num_input > lengthof(group->subtract_input)) {
+							grfmsg(1, "NewSpriteGroup: Industry production wants more inputs than possible, given=%d, max=%d, skipping", group->num_input, lengthof(group->subtract_input));
+							break;
+						}
+						for (uint i = 0; i < group->num_input; i++) {
+							/* XXX: maybe check for same cargo id used multiple times and warn/error? */
```

Comment:
I think it is useful to allow passing 255 here, to skip cargotypes.

Reply:
If you don't want a cargo type to participate in a production callback, don't include it in the parameters to the callback, I don't see what a "skip" value would do.